src/fsevents/fsevents_stubs.c

```diff
@@ -48,33 +55,37 @@ static void dune_fsevents_t_finalize(value v_t) {
 }
 
 static struct custom_operations dune_fsevents_t_ops = {
-    "dune.fsevents.fsevents_t", dune_fsevents_t_finalize,
-    custom_compare_default,     custom_hash_default,
-    custom_serialize_default,   custom_deserialize_default,
-    custom_compare_ext_default, custom_fixed_length_default};
+    "build.dune.fsevents.fsevents_t", dune_fsevents_t_finalize,
+    custom_compare_default,           custom_hash_default,
+    custom_serialize_default,         custom_deserialize_default,
+    custom_compare_ext_default,       custom_fixed_length_default};
 
-CAMLprim value dune_fsevents_runloop_current(value v_unit) {
+CAMLprim value dune_fsevents_dispatch_queue_create(value v_unit) {
   CAMLparam1(v_unit);
-  dune_runloop *rl;
-  rl = caml_stat_alloc(sizeof(dune_runloop));
-  rl->runloop = CFRunLoopGetCurrent();
-  rl->v_exn = Val_unit;
-  caml_register_global_root(&rl->v_exn);
-  value v_runloop = caml_alloc_custom(&dune_fsevents_runloop_ops,
-                                      sizeof(dune_runloop *), 0, 1);
-  Runloop_val(v_runloop) = rl;
-  CAMLreturn(v_runloop);
-}
-
-CAMLprim value dune_fsevents_runloop_run(value v_runloop) {
-  CAMLparam1(v_runloop);
+  dune_dispatch_queue *dq;
+  dq = caml_stat_alloc(sizeof(dune_dispatch_queue));
+  pthread_mutex_init(&dq->dq_lock, NULL);
+  pthread_cond_init(&dq->dq_finished, NULL);
+  dq->dq = dispatch_queue_create("build.dune.fsevents", DISPATCH_QUEUE_SERIAL);
+  dq->v_exn = Val_unit;
+  caml_register_global_root(&dq->v_exn);
+  value v_dq = caml_alloc_custom(&dune_fsevents_dispatch_queue_ops,
+                                 sizeof(dune_dispatch_queue *), 0, 1);
+  Dispatch_queue_val(v_dq) = dq;
+  CAMLreturn(v_dq);
+}
+
+CAMLprim value dune_fsevents_dispatch_queue_run(value v_dq) {
```

Comment:
It looks like this function now just blocks via a condition variable until an exception is raised by the event handler? Previously it had the effect of running the runloop but now it looks like all it does is wait until an exception and then clean up. Perhaps this function's name should reflect the fact that it no longer "runs" anything.

Reply:
I updated the function name to `wait_until_stopped`.